### DISCUSSION TITLE
Build source code problem  cannot convert from 'mshadow::Stream<Device> *' to 'mshadow::Stream<mshadow::cpu> *'

### DISCUSSION THREAD

**Discussion by MonsterPeng** [apache/incubator-mxnet#8772] [Q&A]
I try to build source code on windows server2012 R2. I have got this error.
error C2440: 'default argument' : cannot convert from 'mshadow::Stream<Device> *' to 'mshadow::Stream<mshadow::cpu> *'	

what does it mean?

**Comment by huangyingsong** [comment]
Do u install Microsoft Visual C++ Compiler Nov 2013 CTP? https://mxnet.incubator.apache.org/get_started/windows_setup.html

**Comment by cjolivier01** [comment]
Can you attach full output?  Doesn't show file and line.

**Comment by b0noI** [comment]
proposed labels: "Build", "C++", "Windows"

**Comment by tianxingyzxq** [comment]
same problem on win7 cuda8.0!
![err_mxnet](https://user-images.githubusercontent.com/5278803/37596698-fed31e06-2bb7-11e8-98b5-f7a0d3701b3d.PNG)

**Comment by tianxingyzxq** [comment]
错误	394	error C2440: “默认参数”: 无法从“mshadow::Stream<Device> *”转换为“mshadow::Stream<mshadow::cpu> *”	e:\incubator-mxnet\src\operator\linalg_impl.h	522	1	mxnet
错误	395	error C2440: “默认参数”: 无法从“mshadow::Stream<Device> *”转换为“mshadow::Stream<mshadow::cpu> *”	e:\incubator-mxnet\src\operator\linalg_impl.h	523	1	mxnet
错误	396	error C2440: “默认参数”: 无法从“mshadow::Stream<Device> *”转换为“mshadow::Stream<mshadow::cpu> *”	e:\incubator-mxnet\src\operator\linalg_impl.h	616	1	mxnet
错误	397	error C2440: “默认参数”: 无法从“mshadow::Stream<Device> *”转换为“mshadow::Stream<mshadow::cpu> *”	e:\incubator-mxnet\src\operator\linalg_impl.h	617	1	mxnet
错误	398	error C2664: “std::vector<testing::internal::string,std::allocator<_Ty>>::vector(std::initializer_list<std::basic_string<char,std::char_traits<char>,std::allocator<char>>>,const std::allocator<_Ty> &)”: 无法将参数 1 从“std::string”转换为“const std::allocator<_Ty> &”	E:\incubator-mxnet\dmlc-core\test\unittest\unittest_json.cc	34	1	dmlc_unit_tests
错误	400	error C2039: “isspace”: 不是“std”的成员	E:\incubator-mxnet\example\image-classification\predict-cpp\image-classification-predict.cc	49	1	image-classification-predict
错误	403	error C2783: “std::shared_ptr<_Ty> std::make_shared(_Types &&...)”: 未能为“_Ty”推导 模板 参数	E:\incubator-mxnet\dmlc-core\test\unittest\unittest_lockfree.cc	13	1	dmlc_unit_tests
错误	404	error C2783: “std::shared_ptr<_Ty> std::make_shared(_Types &&...)”: 未能为“_Ty”推导 模板 参数	E:\incubator-mxnet\dmlc-core\test\unittest\unittest_lockfree.cc	14	1	dmlc_unit_tests
错误	450	error C2783: “std::shared_ptr<_Ty> std::make_shared(_Types &&...)”: 未能为“_Ty”推导 模板 参数 (E:\incubator-mxnet\dmlc-core\test\unittest\unittest_thread_group.cc)	E:\incubator-mxnet\dmlc-core\include\dmlc\thread_group.h	631	1	dmlc_unit_tests
错误	490	error LNK1104: 无法打开文件“Debug\libmxnet.lib”	E:\incubator-mxnet\build\LINK	im2rec
